client/components/app.jsx

```diff
@@ -9,26 +9,77 @@ export default class App extends React.Component {
     this.state = {
       view: {
         name: 'catalog',
-        params: {}
+        params: {},
+        cart: []
```

Comment:
The `cart` and `view` states should be separate properties. Be sure to adjust the rest of your code accordingly.